apps/tp_dump.py

```diff
@@ -115,6 +106,166 @@ def tp_channel_histogram(tp_data, quiet=False):
     plt.savefig("tp_channel_histogram.svg")
     plt.close()
 
+def plot_version_histogram(versions, quiet=False):
+    """
+    Plot the TP versions histogram.
+
+    Generally, this will be one value.
+    """
+    plt.figure(figsize=(6,4))
+
+    plt.hist(versions, color='k')
+
+    plt.title("TP Versions Histogram")
+    plt.xlabel("Versions")
+
+    plt.tight_layout()
+    plt.savefig("tp_versions_histogram.svg")
+    plt.close()
+
+def plot_type_histogram(types, quiet=False):
+    """
+    Plot the TP types histogram.
+    """
+    plt.figure(figsize=(6,4))
+
+    plt.hist(types, bins=np.arange(-0.5, 3, 1), range=(0,2), align='mid', color='k')
+
+    plt.title("TP Types Histogram")
+    plt.xlabel("Types")
+    plt.xticks([0,1,2], ["Unknown", "TPC", "PDS"]) # Taken from TriggerPrimitive.hpp
+
+    plt.tight_layout()
+    plt.savefig("tp_types_histogram.svg")
+    plt.close()
+
+def plot_time_start_histogram(times, quiet=False):
+    """
+    Plot the TP time starts histogram.
+    """
+    plt.figure(figsize=(6,4))
+
+    plt.hist(times, color='k')
+
+    plt.title("TP Time Starts Histogram")
+    plt.xlabel("Time Start (Ticks)")
+
+    plt.tight_layout()
+    plt.savefig("tp_time_start_histogram.svg")
+    plt.close()
+
+def plot_time_peak_histogram(times, quiet=False):
+    """
+    Plot the TP time peaks histogram.
+    """
+    plt.figure(figsize=(6,4))
+
+    plt.hist(times, color='k')
+
+    plt.title("TP Time Peaks Histogram")
+    plt.xlabel("Time Peak (Ticks)")
+
+    plt.tight_layout()
+    plt.savefig("tp_time_peak_histogram.svg")
+    plt.close()
+
+def plot_time_over_threshold_histogram(times, quiet=False):
+    """
+    Plot the TP time over threshold histogram.
+    """
+    plt.figure(figsize=(6,4))
+
+    plt.hist(times, color='k')
+
+    plt.title("TP Time Over Threshold Histogram")
+    plt.xlabel("Time Over Threshold (Ticks)")
+
+    plt.tight_layout()
+    plt.savefig("tp_time_over_threshold_histogram.svg")
+    plt.close()
```

Comment:
A lot of these are pretty much identical, with only somewhat different text. Maybe could be worth abstracting this out into one function (not for this PR though)

Reply:
I agree. Since all of these were histograms, very little needed to be changed. Having something that is interactive to only do requested plots and to set their titles, labels, etc. seems like it would be useful, but that would be a future framework function.